packages/request-logic/src/request-logic.ts

```diff
@@ -601,4 +657,86 @@ export default class RequestLogic implements RequestLogicTypes.IRequestLogic {
       this.advancedLogic,
     );
   }
+
+  /**
+   * Computes the diff between the confirmed and pending request
+   *
+   * @param requestStateConfirmed the confirmed request state
+   * @param requestStatePending the pending request state
+   * @returns an object with the pending state attributes that are different from the confirmed one
+   */
+  private computeDiffBetweenPendingAndConfirm(
+    requestStateConfirmed: any,
+    requestStatePending: any,
+  ): Promise<any> {
+    // Compute the diff between the confirmed and pending request
+    let pending: any | null = null;
+    if (!requestStateConfirmed) {
+      pending = requestStatePending;
+    } else if (requestStatePending) {
+      for (const key in requestStatePending) {
+        if (requestStatePending.hasOwnProperty(key)) {
+          // TODO: Should find a better way to do that
+          if (
+            Utils.crypto.normalizeKeccak256Hash(requestStatePending[key]).value !==
+            Utils.crypto.normalizeKeccak256Hash(requestStateConfirmed[key]).value
+          ) {
+            if (!pending) {
+              pending = {};
+            }
+            pending[key] = requestStatePending[key];
+          }
+        }
+      }
+    }
+    return pending;
+  }
+
+  /**
+   * Sorts out the transactions pending older than confirmed ones
```

Comment:
I don't see what "old" pending transactions mean. Do pending transactions have a timestamp ? How can this timestamp be considered safe ?

If it's the case why we want to ignore them ? These old pending transactions will become confirmed and added later to the request anyway.
I don't see any problems to not ignore them since the pending transactions are already computed after the confirmed ones